coreneuron/io/reports/report_configuration_parser.cpp

```diff
@@ -96,14 +113,53 @@ std::vector<ReportConfiguration> create_report_configurations(const char* conf_f
             if (strcmp(report_on, "i_membrane") == 0) {
                 nrn_use_fast_imem = true;
                 report.type = IMembraneReport;
-            } else if (is_soma)
-                report.type = SomaReport;
-            else
-                report.type = CompartmentReport;
+            } else {
+                switch (target_type) {
+                    case TargetType::Soma:
+                        report.type = SomaReport;
+                        break;
+                    case TargetType::Compartment:
+                        report.type = CompartmentReport;
+                        break;
+                    case TargetType::Axon:
+                        report.type = SectionReport;
+                        report.section_type = Axon;
+                        report.section_all_compartments = false;
+                        break;
+                    case TargetType::AxonComp:
+                        report.type = SectionReport;
+                        report.section_type = Axon;
+                        report.section_all_compartments = true;
+                        break;
+                    case TargetType::Dendrite:
+                        report.type = SectionReport;
+                        report.section_type = Dendrite;
+                        report.section_all_compartments = false;
+                        break;
+                    case TargetType::DendriteComp:
+                        report.type = SectionReport;
+                        report.section_type = Dendrite;
+                        report.section_all_compartments = true;
+                        break;
+                    case TargetType::Apical:
+                        report.type = SectionReport;
+                        report.section_type = Apical;
+                        report.section_all_compartments = false;
+                        break;
+                    case TargetType::ApicalComp:
+                        report.type = SectionReport;
+                        report.section_type = Apical;
+                        report.section_all_compartments = true;
+                        break;
+                    default:
+                        std::cerr << "Report error: unsupported target type" << std::endl;
+                        abort();
```

Comment:
we should be as consistent as possible
```suggestion
                        nrn_abort(1);
```

Reply:
I agree... But I used `abort()` because there are other places in that file using it, and not `nrn_abort()`.